cookbooks/redis/recipes/default.rb

```diff
@@ -17,27 +17,27 @@
     end
 
     # Download Redis, if hasn't been downloaded yet
-    remote_file "/data/redis-#{redis_version}.tar.gz" do
+    remote_file "/opt/redis-#{redis_version}.tar.gz" do
       source "#{redis_url}"
       owner node[:owner_name]
       group node[:owner_name]
       mode 0644
       backup 0
-      not_if { FileTest.exists?("/data/#{redis_source_directory}") }
+      not_if { FileTest.exists?(redis_installer_directory) }
     end
 
-    execute "unarchive Redis source" do
-      command "cd /data && tar zxf redis-#{redis_version}.tar.gz && sync"
-      not_if { FileTest.directory?("/data/#{redis_source_directory}") }
+    execute "unarchive Redis installer" do
+      command "cd /opt && tar zxf redis-#{redis_version}.tar.gz && sync"
```

Comment:
In Chef it's generally better to use the `cwd '/opt/my/dir` then to do a `cd /opt/my/dir && cmd`

```
execute "unarchive Redis installer" do
  cwd "/opt"
  command "tar zxf redis-#{redis_version}.tar.gz && sync"
  ...
end